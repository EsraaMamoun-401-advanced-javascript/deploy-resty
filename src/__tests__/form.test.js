import React from 'react';
import { shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import Form from '../components/form/form';
import Results from '../components/results/results';

describe('<Form/>', ()=> {
    it('is alive at application start', () =>{
        let app = shallow(<Form />);
        expect(app.find('span').exists()).toBeTruthy();
        expect(app.find('section').exists()).toBeFalsy();
        expect(app.find('input').exists()).toBeTruthy();
        expect(app.find('label').exists()).toBeTruthy();
        expect(app.find('radio').exists()).toBeFalsy();
    });

    it('changes state on submit', ()=> {
        let app = mount(<Form />);
        let button = app.find('button');
        button.simulate('click');
        expect(app.state('url')).toBe('');
        expect(app.state('method')).toBe('');
    });

    it('renders form correctly', ()=> {
        const formTree = renderer.create(<Form />).toJSON();
        expect(formTree).toMatchSnapshot();
    });

    it('renders results correctly', ()=> {
        const resultsTree = renderer.create(<Results />).toJSON();
        expect(resultsTree).toMatchSnapshot();
    });

});
