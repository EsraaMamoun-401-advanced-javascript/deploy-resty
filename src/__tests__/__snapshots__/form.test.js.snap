// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders form correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    <span>
      URL: 
    </span>
    <input
      name="url"
      onChange={[Function]}
      type="text"
    />
    <button
      type="submit"
    >
      GO!
    </button>
  </label>
  <label
    className="methods"
  >
    <span
      className=""
      id="get"
      onClick={[Function]}
    >
      GET
    </span>
    <span
      className=""
      id="post"
      onClick={[Function]}
    >
      POST
    </span>
    <span
      className=""
      id="put"
      onClick={[Function]}
    >
      PUT
    </span>
    <span
      className=""
      id="delete"
      onClick={[Function]}
    >
      DELETE
    </span>
  </label>
</form>
`;

exports[`<Form/> renders results correctly 1`] = `
<div>
  <h3>
    Headers: 
  </h3>
  <div
    className="react-json-view"
    style={
      Object {
        "backgroundColor": "#272822",
        "cursor": "default",
        "fontFamily": "monospace",
        "position": "relative",
      }
    }
  >
    <div
      className="pretty-json-container object-container"
    >
      <div
        className="object-content"
      >
        <div
          className="object-key-val"
        >
          <span>
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                }
              }
            >
              <div
                className="icon-container"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "17px",
                  }
                }
              >
                <span
                  className="collapsed-icon"
                >
                  <svg
                    fill="currentColor"
                    style={
                      Object {
                        "color": "#ae81ff",
                        "height": "1em",
                        "paddingLeft": "2px",
                        "verticalAlign": "top",
                        "width": "1em",
                      }
                    }
                    viewBox="0 0 15 15"
                  >
                    <path
                      d="M0 14l6-6-6-6z"
                    />
                  </svg>
                </span>
              </div>
              <span
                style={
                  Object {
                    ":hover": Object {
                      "opacity": "1",
                    },
                    "color": "#f9f8f5",
                    "display": "inline-block",
                    "fontStyle": "none",
                    "letterSpacing": "0.5px",
                    "opacity": "0.85",
                    "verticalAlign": "top",
                  }
                }
              >
                <span
                  className="object-key"
                >
                  <span
                    style={
                      Object {
                        "verticalAlign": "top",
                      }
                    }
                  >
                    "
                  </span>
                  <span>
                    root
                  </span>
                  <span
                    style={
                      Object {
                        "verticalAlign": "top",
                      }
                    }
                  >
                    "
                  </span>
                </span>
                <span
                  style={
                    Object {
                      "color": "#f9f8f5",
                      "display": "inline-block",
                      "margin": "0px 5px",
                      "verticalAlign": "top",
                    }
                  }
                >
                  :
                </span>
              </span>
              <span
                style={
                  Object {
                    "color": "#f9f8f5",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "fontWeight": "bold",
                  }
                }
              >
                {
              </span>
            </span>
          </span>
          <span
            className="brace-row"
          >
            <span
              style={
                Object {
                  "color": "#f9f8f5",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontWeight": "bold",
                  "paddingLeft": "0px",
                }
              }
            >
              }
            </span>
            <div
              className="object-meta-data"
              onClick={[Function]}
              style={
                Object {
                  "display": "inline-block",
                  "padding": "0px 0px 0px 10px",
                }
              }
            >
              <span
                className="object-size"
                style={
                  Object {
                    "borderRadius": "3px",
                    "color": "#a59f85",
                    "cursor": "default",
                    "fontStyle": "italic",
                    "margin": "0px 6px 0px 0px",
                  }
                }
              >
                0
                 item
                s
              </span>
              <span
                className="copy-to-clipboard-container"
              >
                <span
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                      "display": "inline",
                    }
                  }
                >
                  <span
                    className="copy-icon"
                  >
                    <svg
                      fill="currentColor"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": "#cc6633",
                          "fontSize": "15px",
                          "height": "1em",
                          "marginRight": "3px",
                          "verticalAlign": "top",
                          "width": "1em",
                        }
                      }
                      viewBox="0 0 40 40"
                    >
                      <g>
                        <path
                          d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                        />
                      </g>
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
  <h3>
    Results: 
  </h3>
  <div
    className="react-json-view"
    style={
      Object {
        "backgroundColor": "#272822",
        "cursor": "default",
        "fontFamily": "monospace",
        "position": "relative",
      }
    }
  >
    <div
      className="pretty-json-container object-container"
    >
      <div
        className="object-content"
      >
        <div
          className="object-key-val"
        >
          <span>
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                }
              }
            >
              <div
                className="icon-container"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "17px",
                  }
                }
              >
                <span
                  className="collapsed-icon"
                >
                  <svg
                    fill="currentColor"
                    style={
                      Object {
                        "color": "#ae81ff",
                        "height": "1em",
                        "paddingLeft": "2px",
                        "verticalAlign": "top",
                        "width": "1em",
                      }
                    }
                    viewBox="0 0 15 15"
                  >
                    <path
                      d="M0 14l6-6-6-6z"
                    />
                  </svg>
                </span>
              </div>
              <span
                style={
                  Object {
                    ":hover": Object {
                      "opacity": "1",
                    },
                    "color": "#f9f8f5",
                    "display": "inline-block",
                    "fontStyle": "none",
                    "letterSpacing": "0.5px",
                    "opacity": "0.85",
                    "verticalAlign": "top",
                  }
                }
              >
                <span
                  className="object-key"
                >
                  <span
                    style={
                      Object {
                        "verticalAlign": "top",
                      }
                    }
                  >
                    "
                  </span>
                  <span>
                    root
                  </span>
                  <span
                    style={
                      Object {
                        "verticalAlign": "top",
                      }
                    }
                  >
                    "
                  </span>
                </span>
                <span
                  style={
                    Object {
                      "color": "#f9f8f5",
                      "display": "inline-block",
                      "margin": "0px 5px",
                      "verticalAlign": "top",
                    }
                  }
                >
                  :
                </span>
              </span>
              <span
                style={
                  Object {
                    "color": "#f9f8f5",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "fontWeight": "bold",
                  }
                }
              >
                {
              </span>
            </span>
          </span>
          <span
            className="brace-row"
          >
            <span
              style={
                Object {
                  "color": "#f9f8f5",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontWeight": "bold",
                  "paddingLeft": "0px",
                }
              }
            >
              }
            </span>
            <div
              className="object-meta-data"
              onClick={[Function]}
              style={
                Object {
                  "display": "inline-block",
                  "padding": "0px 0px 0px 10px",
                }
              }
            >
              <span
                className="object-size"
                style={
                  Object {
                    "borderRadius": "3px",
                    "color": "#a59f85",
                    "cursor": "default",
                    "fontStyle": "italic",
                    "margin": "0px 6px 0px 0px",
                  }
                }
              >
                0
                 item
                s
              </span>
              <span
                className="copy-to-clipboard-container"
              >
                <span
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                      "display": "inline",
                    }
                  }
                >
                  <span
                    className="copy-icon"
                  >
                    <svg
                      fill="currentColor"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": "#cc6633",
                          "fontSize": "15px",
                          "height": "1em",
                          "marginRight": "3px",
                          "verticalAlign": "top",
                          "width": "1em",
                        }
                      }
                      viewBox="0 0 40 40"
                    >
                      <g>
                        <path
                          d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                        />
                      </g>
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
